src/circuit/gadgets/constraints.rs

```diff
@@ -922,5 +921,56 @@ mod tests {
             assert_eq!(expected_or2, or2.get_value().unwrap());
             assert!(cs.is_satisfied());
         }
+
+        #[test]
+        fn prop_enforce_implication_lc_simple(premise_val in any::<bool>(), lc_val in any::<bool>()) {
+            let mut cs = TestConstraintSystem::<Fr>::new();
+
+            let mut lc = LinearCombination::zero();
+            lc = lc + (if lc_val { Fr::one() } else { Fr::zero() }, TestConstraintSystem::<Fr>::one());
+
+            // Allocate premise boolean.
+            let premise = Boolean::from(AllocatedBit::alloc(
+                cs.namespace(|| "premise"),
+                Some(premise_val)
+            ).expect("alloc failed"));
+
+            // Execute the function under test.
+            let _ = enforce_implication_lc(
+                cs.namespace(|| "enforce_implication_lc"),
+                &premise,
+                |_| lc,
+            ).expect("enforce_implication_lc failed");
+
+
+            prop_assert!((!premise_val || lc_val) == cs.is_satisfied())
+        }
+
```

Comment:
Great. Now, if we configure proptest to run this only four times, will it deduplicate cases and therefore guarantee exhaustive testing? Is it smart enough to do this all by itself? My guess is 'no' since we don't use a strategy for the binary LC, but *maybe* if we did?